vm/src/executor.rs

```diff
@@ -229,15 +231,46 @@ pub fn execute(
             }
             Instruction::ChkSig => {
                 let pubkey = Public::from_slice(stack.pop()?.assert_len(64)?.as_ref());
-                let tx_hash = tx.hash_partially(stack.pop()?.as_ref().to_vec(), cur, burn)?;
-                println!("{:?}", tx_hash);
+                let tag = Tag::try_new(stack.pop()?.as_ref().to_vec())?;
+                let tx_hash = tx.hash_partially(tag, cur, burn)?;
                 let signature = Signature::from(Signature::from(stack.pop()?.assert_len(SIGNATURE_LENGTH)?.as_ref()));
                 let result = match verify(&pubkey, &signature, &tx_hash) {
                     Ok(true) => 1,
                     _ => 0,
                 };
                 stack.push(Item(vec![result]))?;
             }
+            Instruction::ChkMultiSig => {
+                // Get n pubkey. If there are more than six pubkeys, return error.
+                let n = stack.pop()?.assert_len(1)?.as_ref()[0] as usize;
+                let mut pubkey: Vec<Public> = Vec::new();
+                for _ in 0..n {
+                    pubkey.push(Public::from_slice(stack.pop()?.assert_len(64)?.as_ref()));
+                }
+
+                // Get m signature. If signatures are more than pubkeys, return error.
+                let m = stack.pop()?.assert_len(1)?.as_ref()[0] as usize;
+                if m > n || m > 6 {
+                    return Err(RuntimeError::InvalidArg)
+                }
+                let mut signatures: Vec<Signature> = Vec::new();
+                for _ in 0..m {
+                    signatures.push(Signature::from(stack.pop()?.assert_len(SIGNATURE_LENGTH)?.as_ref()));
+                }
+
+                let tag = Tag::try_new(stack.pop()?.as_ref().to_vec())?;
+                let tx_hash = tx.hash_partially(tag, cur, burn)?;
+
+                let mut result = 1;
+                while let Some(sig) = signatures.pop() {
```

Comment:
Is it safe when the vector is empty?

How about disallowing `m=0` case?

Reply:
Okay, I thought that in the `m=0`case, it is a right behaviour to success verifying.